specs/AppLifecycle/StateNotifications/AppLifecycle StateNotifications.md

```diff
@@ -0,0 +1,247 @@
+-	Rich activation objects
+-	Selective single/multi-instancing
+-	System state and power notifications.
+
+This spec addresses the System State and Power Notifications APIs in the component.
+
+<br>
+
+## Background
+There are several existing APIs which apps can use to detect changes in system state, in order to participate in improved power usage and battery life. These APIs cover most  if not all of the battery/power state change scenarios that apps would care about - but there's no single API that is easily consumable by all app types. There's also no single API that's undocked from the OS. 
+
+This spec describes new APIs that address these issues by incorporating a comprehensive set of the [Windows.System.Power.PowerManager](https://docs.microsoft.com/en-us/uwp/api/Windows.System.Power.PowerManager) WinRT API in the Reunion AppLifecycle component, and include additional low-level notifications based on the Win32 [PowerSettingRegisterNotification](https://docs.microsoft.com/en-us/windows/win32/api/powersetting/nf-powersetting-powersettingregisternotification) and matching [PowerSettingUnregisterNotification](https://docs.microsoft.com/en-us/windows/win32/api/powersetting/nf-powersetting-powersettingunregisternotification) APIs.
+
+<br>
+
+## Description
+Providing a Reunion version of PowerManager is relatively straightforward, in that we aim to bring across the entire existing API in its current form. For the Win32 APIs, the situation is a little murkier. It boils down to a choice between 2 usage patterns:
+
+- [RegisterPowerSettingNotification](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-registerpowersettingnotification) uses a Windows message-based model. When the app registers for notifications, the system will send out WM_POWERBROADCAST messages.
+- [PowerSettingRegisterNotification](https://docs.microsoft.com/en-us/windows/win32/api/powersetting/nf-powersetting-powersettingregisternotification) uses a callback model. When the app registers for notifications, it supplies a pointer to a callback function, and the system will call back to this function instead of sending messages.
+
+Classic Win32 apps can use either the WM model and/or the callback model. The callback model is more consistent with other app types - including not only modern UWP apps, but also traditional MFC, Winforms and WPF apps. Given this, since all app types can work with the callback model, but only classic Win32 apps use the WM model, the Reunion API will use the callback model for Cobalt. Additional WM-based behavior can be added in a later release, if customer demand indicates it would provide significant benefit.
```

Comment:
Might also call out that the callback model does not prescribe a windowing or threading model, which is better for headless/background components.

Reply:
Updated.